src/NCalc.Core/Parser/LogicalExpressionParser.cs

```diff
@@ -479,7 +407,78 @@ static LogicalExpressionParser()
         Parser = enableParserCompilation ? expressionParser.Compile() : expressionParser;
     }
 
-    private static LogicalExpression GetBinaryExpression((LogicalExpression, IReadOnlyList<(BinaryExpressionType, LogicalExpression)>) x)
+    private static LogicalExpression ParseNumber(ParseContext context, (long, int, long?, long?) number)
+    {
+        long integralValue = number.Item1;
+        int zeroCount = number.Item2;
+        long? decimalPart = number.Item3;
+        long? exponentPart = number.Item4;
+
+        if (((LogicalExpressionParserContext)context).Options.HasFlag(ExpressionOptions.DecimalAsDefault))
+        {
+            decimal result1 = integralValue;
+
+            // decimal part?
+            if (decimalPart != null && decimalPart.Value != 0)
+            {
+                var digits = Math.Floor(Math.Log10(decimalPart.Value) + 1) + zeroCount;
+                result1 += decimalPart.Value / (decimal)Math.Pow(10, digits);
+            }
+
+            // exponent part?
+            if (exponentPart == null) return new ValueExpression(result1);
+            var left = BigDecimal.Parse((double)result1);
```

Comment:
I did a bench, and this seems to be 2x slower than `var left = new BigDecimal(result1);`

| Method    | Mean     | Error     | StdDev    | Ratio | RatioSD | Gen0   | Allocated | Alloc Ratio |
|---------- |---------:|----------:|----------:|------:|--------:|-------:|----------:|------------:|
| TestNew   | 1.390 us | 0.0166 us | 0.0227 us |  1.00 |    0.02 | 0.3395 |     536 B |        1.00 |
| TestParse | 2.339 us | 0.0126 us | 0.0105 us |  1.68 |    0.03 | 0.7133 |    1120 B |        2.09 |